pagopa/io-app
# chore: [IOCOM-1424,IOCOM-1428] FIMS flow cancelling

## Short description
This PR implements the FIMS flow cancelling, either by direct user cancel or by consents denial.

## List of changes proposed in this pull request
- New reducer state value: `abort`. This state represents an asynchronous cancelling/aborting operation that may take time to complete (i.e., when the user cancels having to inform the server with the `abort` endpoint)
- Header’s back button, Abort button and Android physical/virtual back button all invoke a common cancelling method
- The cancelling method dispatch a cancelling action if the state is not `abort`
- New saga listening to the cancelling action. Such saga cancels all running requests and, if consents data are defined, invoke the proper backend API to abort the process
- Both cancelling saga and standard one invoke a common deallocation method at their ends, where all cookies are removed, native resources are deallocated and the navigation pop the stack one time
- The FIMS screen displays a “cancelling” feedback when the abort state is in progress
- main saga split in multiple file sagas
- swipe gesture in react navigation has been disabled in order to avoid any unhandled back navigation
- `ConsentData` type updated to latest specs

## How to test
Using the io-dev-api-server, start a FIMS flow and cancel using:
- the header's back button
- the "Abort" button
- the Android physical/virtual back button

Also check that the standard flow is still working.
